test/plugins/test_convert.py

```diff
@@ -335,3 +335,59 @@ def test_transcode_from_lossy_prevented(self):
             self.run_convert_path(item.path)
         converted = os.path.join(self.convert_dest, b"converted.ogg")
         self.assertNoFileTag(converted, "mp3")
+
+
+@_common.slow_test()
+class NoConvertTest(ConvertTestCase, ConvertCommand):
+    """Test the effect of the `no_convert` option."""
+
+    def setUp(self):
+        super().setUp()
+
+        self.convert_dest = os.path.join(self.temp_dir, b"convert_dest")
+        self.config["convert"] = {
+            "dest": self.convert_dest,
+            "no_convert": "format:OGG",
+            "paths": {"default": "converted"},
+            "format": "mp3",
+            "formats": {
+                "mp3": self.tagged_copy_cmd("mp3"),
+            },
+        }
+
+    def test_no_transcode_when_no_convert_set(self):
+        [item] = self.add_item_fixtures(ext="ogg")
+        with control_stdin("y"):
+            self.run_convert_path(item.path)
+        converted = os.path.join(self.convert_dest, b"converted.mp3")
+        self.assertNotExists(converted)
+
+    def test_no_transcode_when_and_query_no_convert_set(self):
+        self.config["convert"]["no_convert"] = "format:OGG bitrate:..256"
+        [item] = self.add_item_fixtures(ext="ogg")
+        item.bitrate = 128
+        item.store()
+        with control_stdin("y"):
+            self.run_convert_path(item.path)
+        converted = os.path.join(self.convert_dest, b"converted.mp3")
+        self.assertNotExists(converted)
+
+    def test_transcode_when_and_query_no_convert_set_partial_match(self):
+        self.config["convert"]["no_convert"] = "format:OGG bitrate:..256"
+        [item] = self.add_item_fixtures(ext="ogg")
+        item.bitrate = 320
+        item.store()
+        with control_stdin("y"):
+            self.run_convert_path(item.path)
+        converted = os.path.join(self.convert_dest, b"converted.mp3")
+        self.assertExists(converted)
+
+    def test_no_transcode_when_or_query_no_convert_set_partial_match(self):
+        self.config["convert"]["no_convert"] = "format:OGG , bitrate:..256"
+        [item] = self.add_item_fixtures(ext="ogg")
+        item.bitrate = 320
+        item.store()
+        with control_stdin("y"):
+            self.run_convert_path(item.path)
+        converted = os.path.join(self.convert_dest, b"converted.mp3")
+        self.assertNotExists(converted)
```

Comment:
There is a lot of duplication and boilerplate in this test class. This makes it hard to tell what specifically is being tested and the difference between each test case.

Given that you're only interested in the `no_convert` configuration logic, it's best to isolate it to a separate function (see my suggestion above) and create a focused unit test, for example:

```suggestion
class TestNoConvert:
    """Test the effect of the `no_convert` option."""

    @pytest.mark.parametrize(
        "config_value, should_skip",
        [
            ("", False),
            ("bitrate:320", False),
            ("bitrate:320 format:ogg", False),
            ("bitrate:320 , format:ogg", True),
        ],
    )
    def test_no_convert_skip(self, config_value, should_skip):
        item = Item(format="ogg", bitrate=256)

        convert.config["convert"]["no_convert"] = config_value

        assert convert.in_no_convert(item) == should_skip
```

Reply:
Thanks a lot for this that is much nicer, I have created the `in_no_convert` function and updated the tests 👍